anta/tests/system.py

```diff
@@ -299,3 +303,76 @@ def test(self) -> None:
         else:
             data = command_output.split("\n")[0]
             self.result.is_failure(f"The device is not synchronized with the configured NTP server(s): '{data}'")
+
+
+class VerifyNTPAssociations(AntaTest):
+    """Verifies the Network Time Protocol (NTP) associations.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the Primary NTP server should be preferred and will have condition 'sys.peer' and other servers will have condition "candidate".
+    * Failure: The test will fail if the Primary NTP server is NOT preferred.
+
+    Examples
+    --------
+    ```yaml
+    - VerifyNTPAssociations:
+        ntp_servers:
+            - server_address: 1.1.1.1
+              preferred: True
+            - server_address: 2.2.2.2
+            - server_address: 3.3.3.3
+    ```
+    """
+
+    name = "VerifyNTPAssociations"
+    description = "Verifies the Network Time Protocol (NTP) associations."
+    categories: ClassVar[list[str]] = ["system"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show ntp associations")]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyNTPAssociations test."""
+
+        ntp_servers: list[NTPServer]
+        """List of NTP servers."""
+
+        class NTPServer(BaseModel):
+            """Model for a NTP server."""
+
+            server_address: IPv4Address
```

Comment:
If a user configuring the pool then they wont aware about the ip address. So instant of checking the condition on basis of peer address can we check the peers.
```
leaf1-dc1(config)#show ntp associations | json
{
    "peers": {
        "192.168.66.21": {
            "condition": "reject",
            "peerIpAddr": "192.168.66.21",
            "refid": ".INIT.",
            "stratumLevel": 16,
            "peerType": "unicast",
            "lastReceived": 1721650565.0,
            "pollInterval": 1024,
            "reachabilityHistory": [
                false
            ],
            "delay": 0.0,
            "offset": 0.0,
            "jitter": 0.0
        },
        "ntp3.aristanetworks.com": {
            "condition": "sys.peer",
            "peerIpAddr": "10.41.194.20",
            "refid": "17.253.16.125",
            "stratumLevel": 2,
            "peerType": "unicast",
            },
    }
```
In that case server address can be str or IPv4Address. 
@carl-baillargeon @gmuloc